### PR TITLE
update worksheet_util for ace-navigation

### PR DESCRIPTION
**Pull request by pujun-ai**
@percyliang @kashizui This is almost finalized but would still say that is a work in progress since I have not fully tested out the `% search` and `% w_search`.

### CODE REVIEW

File: codalab/lib/worksheet_util.py [hunk 1]
```diff
@@ -589,6 +589,12 @@ def interpret_items(schemas, items):
     # Used to intelligently place the cursor at the focused worksheet item while entering the 'edit' mode.
     interpreted_items_raw_items_map = {}
 
+    raw_items_interpreted_items_list = []
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py]: Document the high-level logic and the invariants.

File: codalab/lib/worksheet_util.py [hunk 2]
```diff
@@ -606,13 +612,13 @@ def get_schema(args):  # args is a list of schema names
     def is_missing(info):
         return 'metadata' not in info
 
-    def flush_bundles():
+    def flush_bundles(start_index, focus_index, sub_focus_index):
         """
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py:655]: Document what this function returns

**Comment by percyliang** [comment]
This would be a good function to have a unit test for.

Also, how about calling it interpreted_to_raw instead of interpreted_items_raw_items_map to make things more concise?

**Comment by pujun-ai** [comment]
@percyliang yeah good point. I have renamed the maps.

File: codalab/lib/worksheet_util.py [hunk 3]
```diff
@@ -749,11 +761,22 @@ def raise_usage_error():
...
 
     def get_command(value_obj):  # For directives only
         return value_obj[0] if len(value_obj) > 0 else None
 
-    last_was_empty_line = False
+    def update_interpreted_raw_map(start_index, end_index, focus_index, sub_focus_index=-1):
+        interpreted_raw_map[str(focus_index) + ',' + str(sub_focus_index)] = end_index
+        raw_interpreted_list.append(((focus_index, sub_focus_index), end_index))
+
+    def update_indexes(end_index, focus_index):
```
Comment (kashizui) [inline codalab/lib/worksheet_util.py]: nit: This function has no side-effects, so `updated_indices` or `next_indices` might make more sense.

**Comment by kashizui** [comment]
LGTM

**Pull request merged**